salt/pillar/postgres.py

```diff
@@ -0,0 +1,117 @@
+# -*- coding: utf-8 -*-
+'''
+Retrieve Pillar data by doing a postgres query
+
+:maturity: new
+:depends: psycopg2
+:platform: all
+
+Complete example
+=====================================
+
+.. code-block:: yaml
+
+    postgres:
+      user: 'salt'
+      pass: 'super_secret_password'
+      db: 'salt_db'
```

Comment:
psycopg2.connection just passes the argument through to libpq.  And that has had support for connection_uri since 9.2

https://www.postgresql.org/docs/9.2/static/libpq-connect.html#LIBPQ-CONNSTRING